forui/lib/src/widgets/switch.dart

```diff
@@ -0,0 +1,177 @@
+  ///
+  /// If set to [DragStartBehavior.start], the drag behavior used to move the
+  /// switch from on to off will begin at the position where the drag gesture won
+  /// the arena. If set to [DragStartBehavior.down] it will begin at the position
+  /// where a down event was first detected.
+  ///
+  /// In general, setting this to [DragStartBehavior.start] will make drag
+  /// animation smoother and setting it to [DragStartBehavior.down] will make
+  /// drag behavior feel slightly more reactive.
+  ///
+  /// By default, the drag start behavior is [DragStartBehavior.start].
+  final DragStartBehavior dragStartBehavior;
+
+  /// Creates a [FSwitch].
+  const FSwitch({
+    required this.value,
+    required this.onChanged,
+    super.key,
+    this.style,
+    this.autofocus = false,
+    this.focusNode,
+    this.onFocusChange,
+    this.dragStartBehavior = DragStartBehavior.start,
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    final style = this.style ?? context.theme.switchStyle;
+    return CupertinoSwitch(
+      value: value,
+      onChanged: onChanged,
+      activeColor: style.checked,
+      trackColor: style.unchecked,
+      thumbColor: style.thumb,
+      focusColor: style.focus,
+      autofocus: autofocus,
+      focusNode: focusNode,
+      onFocusChange: onFocusChange,
+      dragStartBehavior: dragStartBehavior,
+    );
+  }
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties
+      ..add(DiagnosticsProperty('value', value))
+      ..add(DiagnosticsProperty('autofocus', autofocus))
+      ..add(DiagnosticsProperty('dragStartBehavior', dragStartBehavior))
+      ..add(DiagnosticsProperty('style', style))
+      ..add(DiagnosticsProperty<ValueChanged<bool>>('onChanged', onChanged))
+      ..add(DiagnosticsProperty<FocusNode>('focusNode', focusNode))
+      ..add(DiagnosticsProperty<ValueChanged<bool>>('onFocusChange', onFocusChange));
+  }
+}
+
+/// The style of a [FSwitch].
+final class FSwitchStyle with Diagnosticable {
+
+  /// The color of the switch when it is checked.
+  final Color checked;
+
+  /// The color of the switch when it is unchecked.
+  final Color unchecked;
+
+  /// The color of the switch's thumb.
+  final Color thumb;
+
+  /// The color of the switch when it is focused. Defaults to a slightly transparent [checked] color.
+  final Color focus;
+
+  /// Creates a [FSwitchStyle].
+  const FSwitchStyle({
+    required this.checked,
+    required this.unchecked,
+    required this.thumb,
+    required this.focus,
+  });
+
+  /// Creates a [FSwitchStyle] that inherits its properties from [colorScheme].
+  FSwitchStyle.inherit({required FColorScheme colorScheme})
+      : checked = colorScheme.primary,
+        unchecked = colorScheme.border,
+        thumb = colorScheme.background,
+        focus = HSLColor.fromColor(colorScheme.primary.withOpacity(0.80))
```

Comment:
Do we need to convert this to `HSLColor`?

Reply:
This code is also copied from CupertinoSwitch. There probably is a way to do it but I'm not too sure.